### ISSUE TITLE
Duplicate configuration on where the credentials file is stored - why?

### ISSUE THREAD

**Issue by consideRatio** [yuvipanda/gh-scoped-creds#3]
Do we really need a gitconfig and an explicit declaration of the path name of where to put the credentials? I don't understand why that would be needed.

![image](https://user-images.githubusercontent.com/3837114/151035873-c175f034-911c-40d7-848b-0dd2859dd331.png)

As a user of this project I'm a bit extra cautious since its configuration with a significant security impact, not understanding the the notes in the setup instructions fully, even though they may not be related to me at this point, I become a bit cautious.

---

Oh, is it because github-app-user-auth both relies on a jupyter_server/notebook extension to respond somehow and that may act as a user that doesn't have read/write permissions to a home folder. While, the user running the `github-app-user-auth` CLI will have permission to the home folder? Something like that?

**Comment by yuvipanda** [comment]
So `github-app-user-auth` *writes* the credentials, but `git` needs to be told where to *read* the credentials. That's what the config is for. The `store` in `gitconfig` refers to https://git-scm.com/docs/git-credential-store which is part of `git` itself, and we're telling it where to read the config from when it needs credentials.

We should definitely reword the docs (or make a real RTD site) to make this less scary.

**Comment by consideRatio** [comment]
@yuvipanda ahhh I forgot about the distinction between `git` and `github-app-user-auth`. But hmm, can't `github-app-user-auth` read the `git` configuration as well?

**Comment by yuvipanda** [comment]
Ah, it can definitely read the git config! Let's do that.

There is no way to write out the credentials using git when using files to store it I think.

**Comment by consideRatio** [comment]
Preferred solutions in order:
1. Let `git` write the credentials
2. Let `git` parse its own config and report back where the credentials are stored
3. Let `github-app-auth-user` hardcode how to search for `git` config files and parse them.

- Do you agree this is the preference of implementations to solve this?
- Is there another implementation to consider as well?

**Comment by yuvipanda** [comment]
yeah, I agree wrt preferred solutions.

**Comment by consideRatio** [comment]
Ah, so `--git-credentials-path` always needs to be passed unless the default value (`/tmp/git-credentials-path`) matches what's declared in the `gitconfig` file.

I changed my `gitconfig` to point to `/tmp/github-app-auth-user-git-credentials` because I wanted to be explicit in the same sense as when using an environment variable. Doing that, I also needed to explicitly declare `--git-credentials-path` at all times, but the documentation always spoke about non isolated /tmp folder etc. in non-containerized settings.

I guess after fixing the duplication we won't need to think about needing to explicitly pass `--git-credentials-path` any more so no points in considering how to update the docs about that.
